### PR TITLE
Add Glyph Icon Support

### PR DESCRIPTION
**Pull request by taooceros**
#592

TODO:
- [x] Check Relative Path and convert to absolute path

### CODE REVIEW

**Review by jjw24 — DISMISSED** [review]

File: Flow.Launcher/ViewModel/ResultViewModel.cs [hunk 1]
```diff
@@ -68,17 +68,20 @@ public ResultViewModel(Result result, Settings settings)
                         OnPropertyChanged(nameof(Image));
                     });
 
-                if (Result.Glyph.FontFamily.Contains('/'))
+                if (Result.Glyph is { FontFamily: not null } glyph)
                 {
-                    var fontPath = Result.Glyph.FontFamily;
-                    Glyph = Path.IsPathRooted(fontPath) ? Result.Glyph : Result.Glyph with
+                    if (glyph.FontFamily.Contains('/'))
```
Comment (jjw24) [inline Flow.Launcher/ViewModel/ResultViewModel.cs:73]: is this neccessary? can you get the path always and use it without needing to check if it's a path?

Comment (taooceros) [reply Flow.Launcher/ViewModel/ResultViewModel.cs:73]: It allows direct font name if it is installed in the computer. For example, windows 10+ includes Segoe MDL2 Asset so we may not need to bundle a font.

Comment (jjw24) [reply Flow.Launcher/ViewModel/ResultViewModel.cs:73]: This line is for checking if the string is a directory? If so is there a better way to check this?

Comment (taooceros) [reply Flow.Launcher/ViewModel/ResultViewModel.cs:73]: Yeah I don't know whether there's a better way to do this. Probably checking the extension name is also fine?

File: Flow.Launcher/ViewModel/ResultViewModel.cs [hunk 2]
```diff
@@ -60,33 +61,52 @@ public ResultViewModel(Result result, Settings settings)
...
+                    SetImage,
+                    ImageLoader.DefaultImage,
+                    () =>
+                    {
+                        OnPropertyChanged(nameof(Image));
+                    });
+
+                if (Result.Glyph is { FontFamily: not null } glyph)
+                {
+                    if (glyph.FontFamily.Contains('/'))
```
Comment (jjw24) [inline Flow.Launcher/ViewModel/ResultViewModel.cs:73]:
```suggestion
                    // Checks if it's a system installed font, which does not require path to be provided. 
                    if (glyph.FontFamily.Contains('/'))
```

File: Flow.Launcher/ViewModel/ResultViewModel.cs (hunk 1 above)
Comment (jjw24) [reply Flow.Launcher/ViewModel/ResultViewModel.cs:74]: just added a comment, we can work something out later like a dedicated function for determining if string is folder or file path

**Comment by jjw24** [comment]
@taooceros would you mind resolving conflict please

**Comment by taooceros** [comment]
> @taooceros would you mind resolving conflict please

Sure

File: Flow.Launcher/ViewModel/ResultViewModel.cs [hunk 3]
```diff
@@ -16,24 +17,48 @@ public ResultViewModel(Result result, Settings settings)
             if (result != null)
             {
                 Result = result;
+
+                if (Result.Glyph is { FontFamily: not null } glyph)
+                {
+                    if (glyph.FontFamily.EndsWith(".ttf") || glyph.FontFamily.EndsWith(".otf"))
```
Comment (jjw24) [inline Flow.Launcher/ViewModel/ResultViewModel.cs:23]:
```suggestion
                    // Checks if it's a system installed font, which does not require path to be provided. 
                    if (glyph.FontFamily.EndsWith(".ttf") || glyph.FontFamily.EndsWith(".otf"))
```

**Review by jjw24 — APPROVED** [review]

**Pull request merged**